### ISSUE TITLE
User Interaction flow drawing

### ISSUE THREAD

**Issue by bhaanu77** [Spurrya/Lonqa#5]
@clarkjung @Spurrya @hassanabidpk Please find attached the drawing showing user interaction flow with Lonqa and different technical components in Lonqa (this app for hackathon) involved.
![lonqa_interaction_flow_v0 1_2016_03_13](https://cloud.githubusercontent.com/assets/14994833/13728822/138ce074-e91c-11e5-8bb6-3599903e1eca.png)

With this approach we can do the whole app in just nodeJS. We don't even need the speech to text or text to speech in the initial iteration. 
Once the current scope is completed we can add same functionality in mobile app, which will automatically give user's location, identity etc.

**Prioritized steps for implementation** that can happen in parallel:
Step-1: Hardcode location to one famous shopping mall in USA
Step-2: Develop the foundation of the app using Dialog API
Step-3: Add Natural language classifier
Step-4: Add Google location API
Step-5: Add Google drawing API to overlay user's flag on Google map
Step-6: Android app development (Swaroop + Hassan?)
Step-7: Android app integration to Dialog API, Natural language classifier API (Swaroop + Hassan?)
Step-8: Android app integration to **Android Speech to text** (Google Now?) and sending it to Watson Natural language classsifier API (Swaroop + Hassan?)

@Spurrya you are good at UI work, so you might be interested in doing the user's Flag overlay on the Google map using Google [Overlay API](https://developers.google.com/maps/documentation/javascript/overlays#overlay_types) or [drawing layer API](https://developers.google.com/maps/documentation/javascript/drawinglayer#using_the_library) . Anyway Google Drawing API functionality is optional; using Watson APIs is the highest priority.

**Comment by clarkjung** [comment]
Thank you for the illustration! I have a couple of questions. 

1) How do we define the "funniest" answer? 
2) I am not sure if Natural Language Classifier can properly process a user's answer. Has anyone taken a look at some Natural Language Classifier examples?

**Comment by easten20** [comment]
1) How do we define the "funniest" answer? 

"funniest" answer needs to get evaluated by people, we can't do it automatically. one way to do it is to randomly send older answers (e.g. 10 funny answers) from the DB and let the current user to rate it.

**Comment by easten20** [comment]
2) I am not sure if Natural Language Classifier can properly process a user's answer. Has anyone taken a look at some Natural Language Classifier examples?

In order to process user's answer such as location, shops, we should use entity extraction API, for example, we retrieve the "location" or "person" entity from a sentence.

**Comment by hassanabidpk** [comment]
By No.4 You mean **Google GeoLocation API** or   *_Google Places API ?  *_https://developers.google.com/maps/documentation/geolocation/intro

**Comment by bhaanu77** [comment]
@hassanabidpk You are right we need to use Google GeoLocation API to find the latitude and Longitude of the user. If this looks difficult / time taking, we can just ignore whatever location user gives initially and hardcode some location in the system initially.

We need to use Google Places Search API https://developers.google.com/places/web-service/search#PlaceSearchRequests to search what Flower shops / restaurants are nearby.
